### ISSUE TITLE
Validation on array of ref uses `validate` option on individual items and ignores `required: True`

### ISSUE THREAD

**Issue by jfperren** [typegoose/typegoose#513]
Hi all,

First of all, mandatory big thank you for all the hard work on this great project. It's really saving me a ton of time and the syntax is clear and simple, I love it.

Now into the details of this issue: I have a schema where one path is an array of references, and the validation is not working as expected. More specifically, it looks like typegoose is ignoring my `required: True` option and also using my `validate` on the individual elements of the array rather than the array itself.

Here's my class. Basically, I want owners to always be specified and have at least 2 items.

```
export class SharedSomething extends Something {
  @prop({
    ref: () => Owner,
    dim: 1,
    type: mongoose.Types.ObjectId,
    required: true,
    validate: [minArraySize(2)],
  })
  public owners!: Ref<Owner>[];
}
```
`MinArraySize` is implemented in a common file as I reuse it in many places (Btw any chance to have this be a default? I think `minlength` is only for strings right?).
```
export const minArraySize = (minLength: number) => {
  return {
    validator(value): boolean {
      return value.length >= minLength;
    },
    message: `Array should contain at least ${minLength} element`,
  };
};
```

Anyway, with that setup, it seems that two things happen:
1. My `required` gets ignored as I can still save models using `{ owners: undefined }`
2. My `validate` gets applied to the individual elements of the array rather than the array itself. 

I can fix the problem by moving the `validate` option to `@arrayProp`, but given it is now deprecated it doesn't seem like the right solution:
```
export class SharedSomething extends Something {
  @arrayProp({
    required: true,
    validate: [minArraySize(2)],
  })
  @prop({
    ref: () => Owner,
    dim: 1,
    type: mongoose.Types.ObjectId,
    required: true,
  })
  public owners!: Ref<Owner>[];
}
```
Here's the summary of my test cases:
| Test Case | Expectation | using `@prop` | using `@arrayProp` |
|-----|----|----|----|
| `{ owners: [ '<id1>', '<id2>' ] }` | Accepted | ❌ Rejected with error: `AssertionError: expected promise to be fulfilled but it was rejected with 'ValidationError: owners.0: Array should contain at least 2 element, owners.1: Array should contain at least 2 element'` | ✅ Accepted | 
| `{ owners: undefined }` | Reject because `required` | ❌ Accepted | ⚠️Rejected but only because of the `validate` block. If I only use `required: true`, either in `@prop` or @arrayProp`, then it will accept `{ owners: undefined }`.  |
| `{ owners: [] }` | Reject because length < 2 | ❌ Accepted | ✅ Rejected |

I wonder if I am doing anything wrong with my `ref` and `type` here, or if it's a bug in typegoose. If I use `type: () => Owner` it will validate properly but save the `Owner` as an embedded document instead of a reference.

Any clues?

### Version

```
-- @typegoose/typegoose@7.6.0
-- mongoose@5.10.18
```

**Comment by hasezoey** [comment]
> MinArraySize is implemented in a common file as I reuse it in many places (Btw any chance to have this be a default? I think minlength is only for strings right?).

such things should be asked in mongoose, not here

> I can fix the problem by moving the validate option to @arrayProp, but given it is now deprecated it doesn't seem like the right solution:

`@arrayProp` is literally an pass-through to `@prop` with one extra parameter (explicitly setting `WhatIsIt` to `ARRAY`

---

this sounds like tsconfig option `emitDecoratorMetadata` is disabled, which would enable auto-detection of an array based on the typescript type, if enabling dosnt help, or dont want to enable it, try [this guide](https://typegoose.github.io/typegoose/docs/guides/use-without-emitDecoratorMetadata)

**Comment by jfperren** [comment]
Hi @hasezoey, thanks for the quick reply. I am a bit confused, because:
1. `emitDecoratorMetadata` is already set to `true` in my tsconfig
2. If I use `@prop({ ... }, WhatIsIt.ARRAY)` directly, it does not solve my problem.

It seems that the issue is specifically with the `Ref` type, as the following validates correctly:

```

**Comment by hasezoey** [comment]
this should be fixed with b39eff9367e6910515944a140d7c939dddc532e3, since that commit `ref` and `refPath` also use `mapArrayOptions` - and because both types have some of the same options, if you want to force it into something specific, look there:
- for refType options (or inner): https://typegoose.github.io/typegoose/docs/api/decorators/prop#inneroptions
- for array options (or outer): https://typegoose.github.io/typegoose/docs/api/decorators/prop#outeroptions

you can have both defined and non conflicting (`innerOptions` and `outerOptions` if they contain options from auto-mapping, auto-mapping values will be overwritten)
